vllm/model_executor/layers/mamba/ops/mamba_ssm.py

```diff
@@ -344,28 +374,33 @@ def selective_scan_fn(u,
         C = C.contiguous()
     if z is not None and z.stride(-1) != 1:
         z = z.contiguous()
-    if B.dim() == 3:
+    if B.dim() == 3 and cu_seq_len is None:
         B = B.unsqueeze(1)
-    if C.dim() == 3:
+    if B.dim() == 2 and cu_seq_len is not None:
+        B = B.unsqueeze(0)
+    if C.dim() == 3 and cu_seq_len is None:
         C = C.unsqueeze(1)
-    n_chunks = int((u.shape[-1] + 2048 - 1) / 2048)
-    x = torch.zeros((
-        u.shape[0],
-        u.shape[1],
-        n_chunks,
-        int(A.shape[1] * 2),
-    ),
-                    device=u.device,
-                    dtype=torch.float32,
-                    requires_grad=False)
-    x[:, :, 0, 0::2] = 1
-    if prev_state is not None:
-        x[:, :, 0, 1::2].copy_(prev_state)
-    out, x, *rest = ops.selective_scan_fwd(u, delta, A, B, C, D, z, delta_bias,
-                                           delta_softplus, position_indices, x)
-    last_state = x[:, :, -1, 1::2]  # (batch, dim, dstate)
+    if C.dim() == 2 and cu_seq_len is not None:
+        C = C.unsqueeze(0)
+
+    if ssm_states is None:
+        ssm_states = torch.zeros((
+            u.shape[0],
+            u.shape[1],
+            int(A.shape[1]),
+        ),
+                                 device=u.device,
+                                 dtype=u.dtype,
+                                 requires_grad=False)
```

Comment:
nit: this is a little hard to read with the current formatting -- perhaps better to split out the shape definition into its own line

Reply:
Addressed